Fixed issue where item wasn't moved to new section.

jallen opened the pull request:
#### :tophat: What? Why?

If an item is moved to a new section and the section doesn't exist the item is Inserted instead of moved causing the section in the tableView to not be displayed.
#### :family: Dependencies?
#### :clipboard: Developer checklist
- [ ] Documentation
- [x] Unit Tests 
- [ ] Update README (if necessary)
#### :ghost: GIF

![]()
If the section didn't exist before the item was moved the updated item would be inserted instead of moved causing the new section to not be displayed properly in the tableView.

jallen committed:
Fixed issue where item wasn't moved to new section.

If the section didn't exist before the item was moved the updated item would be inserted instead of moved causing the new section to not be displayed properly in the tableView.

jallen committed:
Fixed issue where item wasn't moved to new section.

If the section didn't exist before the item was moved the updated item would be inserted instead of moved causing the new section to not be displayed properly in the tableView.

polqf commented on Example/RealmResultsController-iOSTests/RealmResultsCacheTests.swift:
```diff
@@ -480,7 +485,20 @@ class CacheSpec: QuickSpec {
+                    type = cache.updateType(task)
+                }
+                it("returns type .Move") {
+                    expect(type) == RealmCacheUpdateType.Move
```
I think in this case the desired test to have is if a nonResolvedTask `resolved` flag changes to `true`, it should return a `.Move`. In this concrete case it should return a deletion, as the predicate of the `request` is saying that it only "allows" tasks that have `resolved == true`, and this is not the case

jallen committed:
filter initial objects

Use array filter instead of predicate to limit initial set of tasks to be contained in one section using "resolved" sectionKeyPath.

jallen committed:
filter initial objects

Use array filter instead of predicate to limit initial set of tasks to be contained in one section using "resolved" sectionKeyPath.

polqf committed:
Update carthage frameworks to Swift3.0

polqf committed:
Update Source to Swift3.0

polqf committed:
Update Tests to Swift3.0

polqf committed:
Update travis.yml

polqf committed:
8b3 ABI Mismatch fix

polqf committed:
Remove unused self

polqf committed:
Xcode8b6 Carthage update

polqf committed:
Xcode8b6 Fixes

polqf committed:
Revert "8b3 ABI Mismatch fix"

This reverts commit 9a6156d3f2f9f0eb712b758ceeece081b9de80d6.

polqf committed:
Fix crash on real device

polqf committed:
Update to Realm v1.1.0

polqf committed:
Bump to 0.5.0

polqf committed:
Add .swift-version

polqf committed:
Point to correct Realm version in podspec

polqf committed:
Add CocoaPods disclaimer

polqf committed:
Update Cartfile to use Realm v2.0.2

polqf committed:
Avoid Retain Cycle in RealmLogger

jallen committed:
build for Swift 3.0.2

jallen committed:
update to Swift 3 syntax

jallen committed:
Merge branch 'bug/new-section-move' of github.com:jallen/RealmResultsController into bug/new-section-move

# Conflicts:
#	Example/RealmResultsController-iOSTests/RealmResultsCacheTests.swift
#	Source/RealmResultsCache.swift
